compiler/src/main/scala/org/scalajs/core/compiler/GenJSCode.scala

```diff
@@ -264,6 +307,9 @@ abstract class GenJSCode extends plugins.PluginComponent
           }
         }
 
+        assert(lazilyGeneratedAnonClasses.isEmpty, "Unused anonymous classes: " +
+            lazilyGeneratedAnonClasses.keys.map(_.fullName).mkString(", "))
```

Comment:
This is not a valid check anymore, since a class might be used during `GenMethodAsJSFunction` and not be removed from the map.